crates/wit-component/tests/components/lower-options/component.wat

```diff
@@ -0,0 +1,284 @@
+(component
+  (type (;0;) (func))
+  (type (;1;) (list string))
+  (type (;2;) (func (param "x" (type 1))))
+  (type (;3;) (record (field "s" string)))
+  (type (;4;) (func (param "x" (type 3))))
+  (type (;5;) (variant (case "s" string)))
+  (type (;6;) (func (param "x" (type 5))))
+  (type (;7;) (record (field "s" u32)))
+  (type (;8;) (func (param "x" (type 7))))
+  (type (;9;) (variant (case "s" u32)))
+  (type (;10;) (func (param "x" (type 9))))
+  (type (;11;) (list (type 3)))
+  (type (;12;) (func (param "x" (type 11))))
+  (type (;13;) (list (type 5)))
+  (type (;14;) (func (param "x" (type 13))))
+  (type (;15;) (list u32))
+  (type (;16;) (func (param "x" (type 15))))
+  (type (;17;) (func (param "x" u32)))
+  (type (;18;) (tuple u32 u32))
+  (type (;19;) (func (result (type 18))))
+  (type (;20;) (func (result string)))
+  (type (;21;) (func (result (type 15))))
+  (type (;22;) (func (result u32)))
+  (type (;23;) (func (result (type 5))))
+  (type (;24;) (list (type 9)))
+  (type (;25;) (func (result (type 24))))
```

Comment:
This approach was taken to try to minimize the type definitions and share common ones between the instance types; perhaps a premature optimization and since things like parameter names are encoded in function type definitions, it might not end up sharing all that much.

I'd be totally happy with encoding the type information in each instance type directly rather than aliasing.

Reply:
Ah ok nah that makes sense!